engine/engine.go

```diff
@@ -241,24 +238,26 @@ func (e *EngineAPI) GetPayloadV3(payloadID engine.PayloadID) (*eth.ExecutionPayl
 	e.lock.RLock()
 	defer e.lock.RUnlock()
 
-	payload := e.payloadStore.Current()
-	if currentID := payload.ID(); payloadID != *currentID {
+	if payloadID != *e.payloadAttr.ID() {
 		return nil, engine.InvalidParams.With(errors.New("payload is not current"))
 	}
 
 	// TODO: handle time slot based block production
 	// for now assume block is sealed by this call
 	if err := e.builder.Build(&builder.Payload{
-		Transactions: payload.CosmosTxs,
-		GasLimit:     payload.GasLimit,
-		Timestamp:    payload.Timestamp,
-		NoTxPool:     payload.NoTxPool,
+		Transactions: e.payloadAttr.CosmosTxs,
+		GasLimit:     e.payloadAttr.GasLimit,
+		Timestamp:    e.payloadAttr.Timestamp,
+		NoTxPool:     e.payloadAttr.NoTxPool,
 	}); err != nil {
 		log.Panicf("failed to commit block: %v", err) // TODO error handling. An error here is potentially a big problem.
 	}
-	// TODO remove payload from payload store.
+	payloadEnvelope := e.payloadAttr.ToExecutionPayloadEnvelope(e.blockStore.HeadBlock().Hash())
```

Comment:
Do we need this variable or can we do 
```go
	return payload.ToExecutionPayloadEnvelope(e.blockStore.HeadBlock().Hash()), nil
```
as before?

Reply:
We need to store this temporarily in order to set the payloadAttr to nil before `return`ing. Probably less awkward after the payloadenvelope construction is fixed to improve all block data.